stores/facilitiesStore.ts

```diff
@@ -29,9 +30,67 @@ export const useFacilitiesStore = defineStore(
             // googleMapsFields
             googlemapsURL: ref('') as Ref<string>,
             mapLatitude: ref('') as Ref<string>,
-            mapLongitude: ref('') as Ref<string>
+            mapLongitude: ref('') as Ref<string>,
+            healthcareProfessionalIds: ref([]) as Ref<string[]>
         } as { [key: string]: Ref }
 
+        function setSelectedFacilityData(facilityId: string) {
+            selectedFacilityData.value = facilityData.value
+                .find((facility: Facility) => facility.id === facilityId)
+            initializeFacilitySectionValues(selectedFacilityData.value)
+        }
+
+        /**This function is a helper that checks for if the data passed is a Facility or Submission
+            in order to properly handle the conditional to autofill this section of the form **/
+        function isFacility(data: Facility | Submission): data is Facility {
+            return (data as Submission).facility === undefined
+        }
```

Comment:
We may have to rethink this function a bit. It's possible for it to always return True, because `data.facility` can be `undefined` for both a `Facility` and a `Submission`:
![Screenshot 2024-11-07 at 1 06 14 AM](https://github.com/user-attachments/assets/b0c34eaf-89a0-4a7b-9ed9-7a7be2193900)

Reply:
You are correct. Actually I believe I was overthinking it.

When we eventually use the facility section in the submission that autofill can be in the submission store since I am autofilling based on the path and screen we are on which already is choosing whether it is a submission or existing facility.

I'll cut this and the autofill for submission out of this PR and wait until later when there is more refactoring.